File: examples/serialization/SaveTable.java
import java.awt.Color;
import java.io.FileOutputStream;
import java.io.ObjectOutputStream;
import java.util.Hashtable;

public class SaveTable {
	public static void main(String[] args) {
		Hashtable<String, Object> h = new Hashtable<String, Object>();
		// Hashtable implements the serializable interface so it can be freeze-dried
		h.put("string", "Marco Polo");
		h.put("int", 1619);
		h.put("double", Math.PI);
		h.put("color", new Color(255, 0, 0));

		try {
			FileOutputStream fileout = new FileOutputStream("hash.ser");
			ObjectOutputStream out = new ObjectOutputStream(fileout);
			out.writeObject(h);
			out.close();
		} catch (Exception e) {
			System.out.println(e);
		}
	}
}

File: examples/go/serialization/SaveTable.go
//Author: Amit Jain & Paul Ellis
package main

import (
	"math"
	"image/color"
	"bytes"
	"encoding/gob"
	"os"
)

func CheckErr(e error){
	if e != nil{
		panic(e)
	}
}

func main(){
	pi := math.Pi
	var col color.Color
	col = color.RGBA{255,0,0,0}
	table := make(map[string]any)

	table["string"] = "Marco Polo"
	table["int"] = 1619
	table["float64"] = pi
	table["color"] = col

	var buffer bytes.Buffer

	//If you use any interfaces/structs it's best to register them
	//to the gob so that it can encode correctly.
	gob.Register(color.RGBA{})

	//Create a new encoder that will output 
	//the encoding to the buffer
	enc := gob.NewEncoder(&buffer)

	//Encode the data from table into the buffer
	err := enc.Encode(table)
	CheckErr(err)

	//Write out the buffer's byte slice to hash.ser file. 
	//Uses the last argument to set R/W/E permission
	//This is a one and done file write so there is no need
	//to flush or close a writer
	err = os.WriteFile("hash.ser", buffer.Bytes(), 0644)
	CheckErr(err)

}